src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -943,6 +944,95 @@ private Task ConfigFileDeserializationValidationHelper(string jsonString)
             return Verify(runtimeConfig);
         }
 
+        /// <summary>
+        /// Helper method to parse the config json string and set the connection string field to a given string.
+        /// </summary>
+        /// <param name="configJsonString">Config Json string</param>
+        /// <param name="newConnectionString">Value to update the connection string with</param>
+        /// <returns>Config Json string with connection string set to the given value</returns>
+        private static string UpdateConnectionStringWithCustomValue(string configJsonString, string newConnectionString)
+        {
+            JObject configJson = JObject.Parse(configJsonString);
```

Comment:
Did you rule out usage of `JsonNode` from system.text.json?  https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/use-dom#use-jsonnode